jib-core/src/main/java/com/google/cloud/tools/jib/frontend/MainClassFinder.java

```diff
@@ -36,20 +37,49 @@
   /** Helper for loading a .class file. */
   private static class ClassFileLoader extends ClassLoader {
 
-    private final Path classFile;
+    private Path rootDirectory;
+    private Path classFile;
 
-    private ClassFileLoader(Path classFile) {
+    private ClassFileLoader(Path classFile, Path rootDirectory) {
       this.classFile = classFile;
+      this.rootDirectory = rootDirectory;
     }
 
     @Nullable
     @Override
     public Class findClass(@Nullable String name) {
       try {
+        // Name is only ever null when we call findClass manually, and not null otherwise. If null,
+        // we should resolve the correct filename.
+        if (name != null) {
+          classFile = rootDirectory;
```

Comment:
Wouldn't this break the later calls to `findClass(null)`?

Reply:
Not at the moment, since we only ever call it ourselves after constructing a new `ClassFileLoader`.

Reply:
Yeah, the members should be able to be `private final`